spikeinterface/preprocessing/normalize_scale.py

```diff
@@ -215,17 +227,34 @@ class ZScoreRecording(BasePreprocessor):
     """
     name = 'center'
 
-    def __init__(self, recording, mode="median+mad",
+    def __init__(self, recording, mode="median+mad", gain=None, offset=None, int_scale=None,
                  dtype="float32", **random_chunk_kwargs):
         
         assert mode in ("median+mad", "mean+std")
 
+        # fix dtype
+        dtype_ = fix_dtype(recording, dtype)
+
+        if dtype_.kind == 'i':
+            assert int_scale is not None, 'For recording with dtype=int you must set dtype=float32 OR set a scale' 
+
         random_data = get_random_data_chunks(recording, **random_chunk_kwargs)
 
-        if mode == "median+mad":
+        if gain is not None:
+            assert offset is not None
+            gain = np.asarray(gain)
+            offset = np.asarray(offset)
+            n = recording.get_num_channels()
+            assert gain.shape[0] == n
+            assert offset.shape[0] == n
+            if gain.ndim == 1:
+                gain = gain[None, :]
+            if offset.ndim == 1:
+                offset = offset[None, :]
```

Comment:
Not working otherwise...

```suggestion
            if gain.ndim == 1:
                gain = gain[None, :]
            assert gain.shape[1] == n
            if offset.ndim == 1:
                offset = offset[None, :]
            assert offset.shape[1] == n
```

Reply:
done in last commit